src/main.rs

```diff
@@ -253,6 +230,19 @@ async fn run(miniserve_config: MiniserveConfig) -> Result<(), ContextualError> {
         .map_err(|e| ContextualError::IoError("".to_owned(), e))
 }
 
+/// Allows us to set low-level socket options
```

Comment:
Please add some more comments here with some context as to why we do this, perhaps even link the PR for the discussion in addition as this is a fairly unobvious low-level thing that we have going on here.